regionServiceClient/lib/cloudregister/smt.py

```diff
@@ -0,0 +1,151 @@
+        """Return the fully qualified domain name"""
+        return self._fqdn
+
+    # --------------------------------------------------------------------
+    def get_name(self):
+        """Return the name"""
+        return self._fqdn.split('.', 1)[0]
+
+    # --------------------------------------------------------------------
+    def get_ip(self):
+        """Return the IP address"""
+        return self._ip
+
+    # --------------------------------------------------------------------
+    def is_equivalent(self, smt_server):
+        """When 2 SMT servers have the same cert fingerprint and their
+           FQDN is the same they are equivalent."""
+        if (
+                self.get_FQDN() == smt_server.get_FQDN() and
+                self.get_fingerprint() == smt_server.get_fingerprint()):
+            return 1
+
+        return None
+
+
+    # --------------------------------------------------------------------
+    def is_responsive(self):
+        """Check if the SMT server is responsive"""
+        request = self.__request_cert()
+        if request and request.status_code == 200:
+            if self.__is_cert_valid(request.text):
+                return True
+            msg = 'Cert verify failed during access test, notify administrator'
+            logging.error(msg)
+
+        return False
+
+
+    # --------------------------------------------------------------------
+    def write_cert(self, target_dir):
+        """Write the certificate to the given directory"""
+        logging.info('Writing SMT rootCA: %s' % target_dir)
+        ca_file_path = (target_dir +
+                '/registration_server_%s.pem' % self.get_ip().replace('.','_'))
+        try:
+            smt_ca_file = open(ca_file_path, 'w')
+            smt_ca_file.write(self.get_cert())
+            smt_ca_file.close()
+        except IOError:
+            errMsg = 'Could not store SMT certificate'
+            logging.error(errMsg)
+            return 0
+
+
+        return 1
+
+
+    # Private
+    # --------------------------------------------------------------------
+    def __is_cert_valid(self, cert):
+        """Verfify that the fingerprint of the given cert matches the
+           expected fingerprint"""
+        try:
+            x509 = X509.load_cert_string(str(cert))
+            x509_fingerprint = x509.get_fingerprint('sha1')
+        except:
+            errMsg = 'Could not read X509 fingerprint from cert'
+            logging.error(errMsg)
+            return False
+        
+        if x509_fingerprint != self.get_fingerprint().replace(':', ''):
+            errMsg = 'Fingerprint could not be verified'
+            logging.error(errMsg)
+            return  False
+
+        return True
+
+
+    # --------------------------------------------------------------------
+    def __request_cert(self):
+        """Request the cert from the SMT server and return the request"""
+        cert_rq = None
+        attempts = 0
+        retries = 3
```

Comment:
That would imply that somewhere along the line they can be changed via arguments, which they cannot today, and I am not certain I want to go in that direction. This is only there to accommodate network flakiness of the framework and I really do not want to encourage having a work for a insufficiently responsive framework implementation.